src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1186,9 +1187,9 @@ private static TypeSymbol GetParameterType(int argIndex, MemberAnalysisResult re
                     continue;
                 }
 
-                if (okToDowngradeToNeither)
+                if (!considerRefKinds || Conversions.ClassifyImplicitConversion(type1, type2, ref useSiteDiagnostics).Kind != ConversionKind.Identity)
                 {
-                    Debug.Assert(Conversions.ClassifyImplicitConversion(type1, type2, ref useSiteDiagnostics).Kind != ConversionKind.Identity);
+                    Debug.Assert(considerRefKinds || Conversions.ClassifyImplicitConversion(type1, type2, ref useSiteDiagnostics).Kind != ConversionKind.Identity);
```

Comment:
Can you please make that assumption explicit (e.g. cut and paste your response above after the first sentence).